### PR TITLE
Add google_api_to_s3_transfer example dags and system tests

### PR DESCRIPTION
**Pull request by feluelle**
This PR / change adds google_api_to_s3_transfer systems tests by running the newly added example dags. It also adds a aws system helper which can be used for providing aws context like connection or a s3 bucket for example to the system tests so you do not need to care about how to create or delete resources.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

cc @xinbinhuang @mustafagok - PTAL :)

### CODE REVIEW

File: tests/test_utils/aws_system_helpers.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+@pytest.mark.system("amazon")
+class AmazonSystemTest(SystemTest):
+
+    @classmethod
+    def execute_with_ctx(cls, cmd: List[str]):
+        """
+        Executes command with context created by provide_gcp_context and activated
+        service key.
```
Comment (feluelle) [inline tests/test_utils/aws_system_helpers.py]:
```suggestion
        Executes command with context created by provide_aws_context.
```

File: tests/test_utils/aws_system_helpers.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+from tests.test_utils.system_tests_class import SystemTest
+from tests.utils.logging_command_executor import get_executor
+
+AWS_DAG_FOLDER = os.path.join(
+    AIRFLOW_MAIN_FOLDER, "airflow", "providers", "amazon", "aws", "example_dags"
+)
+
+
+@contextmanager
+def provide_aws_context():
```
Comment (potiuk) [inline tests/test_utils/aws_system_helpers.py]: Nice! I like the way you used context managers here :)

**Review by potiuk — APPROVED** [review]
Love it !

**Comment by potiuk** [comment]
:heart:

**Comment by potiuk** [comment]
There are some problems with test collection though :( see the GA errors..

**Comment by xinbinhuang** [comment]
Awesome! Thanks for creating the `aws_system_helper.py`!

**Comment by feluelle** [comment]
> There are some problems with test collection though :( see the GA errors..

..due to my changes? I committed my code suggestion to check if the errors are persistent.

**Comment by feluelle** [comment]
```
 E   airflow.exceptions.AirflowException: Invalid arguments were passed to BranchPythonOperator (task_id: check_and_transform_video_ids). Invalid arguments were:
E   *args: ()
E   **kwargs: {'provide_context': True}
```
That's because I need `provide_context=True` in backported packages for 1.10.*

@potiuk When I run backport packages script in breeze with airflow version 2.0dev it does not add `{'provide_context': True}` to the backported packages.

How do I get it to work for both 2.0 and 1.10.* ?

**Comment by feluelle** [comment]
Do we currently miss to generate backport packages for core operators which are not included in `providers` like branch operator?

**Comment by potiuk** [comment]
@feluelle  - We don't have (and did not plan to) have backported packages for the core operators. They are far too much "core", we think and tied to the airflow core.

We could reconsider that - but for now, - those operators are so "core" that we left them in the place they are because we think if we change it, it will mean far, far far too many changes across 100% of days out there and we wanted to avoid that.

The way we handle it now - we have the bowler automated refactoring when the packages are being prepared and we do some small refactorings like that. Those are very small changes: 

You can see some examples here:
https://github.com/apache/airflow/blob/master/backport_packages/setup_backport_packages.py#L167 

I think @turbaszek can help with it as he wrote most of the bowler rules.

**Comment by potiuk** [comment]
And @feluelle -> actually, it should be very easy. You do not need to have provide_context in those operators. We already have the rule to add "provide_context" to Python operator automatically via the bowler rule I linked when the packages are prepared so the only change will be to add it to BranchPythonOperator as well.

**Comment by potiuk** [comment]
And  maybe there are other *Python operators as well :)?

**Comment by potiuk** [comment]
@turbaszek -> I just looked at the code - should not that be removed? 

I think we should add provide_context to all *Python operators out there :). No matter where they are used.

```
    .is_filename(include=r"mlengine_operator_utils.py$")  
```

**Comment by feluelle** [comment]
Thanks @potiuk for the information. So would I, for now, just add the file to the `is_filename` expression or should we really run it on all `*PythonOperator`s. PTAL @turbaszek :)

File: backport_packages/setup_backport_packages.py [hunk 3]
```diff
@@ -251,6 +251,13 @@ def _remover(node: LN, capture: Capture, filename: Filename) -> None:
         .modify(add_provide_context_to_python_operator)
     )
 
+    (
+        qry.select_function("BranchPythonOperator")
+        .is_call()
+        .is_filename(include=r"example_google_api_to_s3_transfer_advanced.py$")
+        .modify(add_provide_context_to_python_operator)
+    )
```
Comment (feluelle) [inline backport_packages/setup_backport_packages.py:259]: For now I added this specific case here. If we want to remove the `.is_filename` filter this should be a separate PR. WDYT @potiuk ?

**Comment by mustafagok** [comment]
I see that you have much bigger problems, I would like to suggest a small change, about naming "aws_system_helpers.py" as "amazon_system_helpers.py" since it contains `AmazonSystemTest` class and `@pytest.mark.system("amazon")`

**Comment by feluelle** [comment]
> I see that you have much bigger problems, I would like to suggest a small change, about naming "aws_system_helpers.py" as "amazon_system_helpers.py" since it contains AmazonSystemTest class and @pytest.mark.system("amazon")

Fixed. 👍

**Comment by feluelle** [comment]
@mik-laj or @turbaszek want to have a last look? I would like to merge it (Jarek approved it already) but I also would like to have an opinion on the change in `setup_backport_packages.py`.

File: backport_packages/setup_backport_packages.py (hunk 3 above)
Comment (mik-laj) [reply backport_packages/setup_backport_packages.py:259]: I think it's worth limiting changes and not changing all files automatically. If we need to change another example, it will be easy to add here., ..so LGTM

**Comment by feluelle** [comment]
The docs build failed on master due to missing :doc:`How to use <howto/operator/amazon/aws/google_api_to_s3_transfer>` - follow-up PR is coming.

I am wondering why the build docs task succeeded on this branch but fails on master.

**Comment by mik-laj** [comment]
You didn't do rebase before merging it. This check has been added 3 days ago. https://github.com/apache/airflow/commit/923f4231126b84de90d1e78ce0470c153a67f0ea

**Comment by feluelle** [comment]
Thank you Kamil 👍 :)
